### ISSUE TITLE
[BUG] Managed DHCP service should filter out VMs attaching to the pod network

### ISSUE THREAD

**Issue by starbops** [harvester/harvester#6624]
**Describe the bug**

<img width="1468" alt="image" src="https://github.com/user-attachments/assets/66007890-b85f-418c-80a7-667dad0a3082">

Virtual machines attaching to the `management Network` is essentially on the pod network. The IPAM is taken care by the CNI plug-in. The managed DHCP service is of no use in this scenario. Currently, the vm-dhcp-controller is working correctly, which means it does not allocate IP addresses for those kind of virtual machines, but the VirtualMachineNetworkConfig's status does not reflect the situation appropriately:

```
apiVersion: network.harvesterhci.io/v1alpha1
kind: VirtualMachineNetworkConfig
metadata:
  creationTimestamp: "2024-09-24T07:47:52Z"
  finalizers:
  - wrangler.cattle.io/vm-dhcp-vmnetcfg-controller
  generation: 1
  labels:
    harvesterhci.io/vmName: mgmt-masq
  name: mgmt-masq
  namespace: default
  ownerReferences:
  - apiVersion: kubevirt.io/v1
    kind: VirtualMachine
    name: mgmt-masq
    uid: 2ba42db0-4ac9-494b-bfaa-4baaf8e431bb
  resourceVersion: "927335"
  uid: 35bf7403-29ed-4252-9493-96e72bd5312b
spec:
  vmName: mgmt-masq
status:
  conditions:
  - lastUpdateTime: "2024-09-24T07:47:55Z"
    status: "True"
    type: Allocated
  - lastUpdateTime: "2024-09-24T07:47:55Z"
    status: "False"
    type: Disabled
```

Note that though there's no IP address shown, the `Allocated` condition is still `True`, which is not so true.

**To Reproduce**
1. Create a virtual machine with its network selected as `management Network`

**Expected behavior**

In short, the `Allocated` condition of the VirtualMachineNetworkConfig object should be `False` and give a meaningful `Reason` or `Message`.

**Support bundle**

N/A

**Environment**
 - Harvester ISO version: v1.3.1
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): KVM
 - harvester-vm-dhcp-controller version: v0.3.1

**Additional context**

N/A

**Comment by albinsun** [comment]
Hi @pratikjagrut 
Please help to verify this issue, thank you.

**Comment by pratikjagrut** [comment]
All test cases pass as provided in the test plan in PR.

# Environment
Harvester: 1.6.0-rc2

# Test Scenario

<details>
<summary>Case 1 - Create a VM on the management Network</summary>

<img width="676" height="55" alt="Image" src="https://github.com/user-attachments/assets/3c37c092-d06a-4763-83d6-3a90873aa1f1" />

</details>

<details>
<summary>Case 2 - Manually create a VirtualMachineNetworkConfig CR for the VM</summary>

<img width="643" height="158" alt="Image" src="https://github.com/user-attachments/assets/3e7efbfd-645c-4020-bd98-d03da1e06b4d" />

</details>

**Comment by pratikjagrut** [comment]
All test cases pass as provided in the test plan in PR.

# Environment
Harvester: 1.6.0-rc2

# Test Scenario

<details>
<summary>Case 1 - Create a VM on the management Network</summary>

<img width="676" height="55" alt="Image" src="https://github.com/user-attachments/assets/3c37c092-d06a-4763-83d6-3a90873aa1f1" />

</details>

<details>
<summary>Case 2 - Manually create a VirtualMachineNetworkConfig CR for the VM</summary>

<img width="643" height="158" alt="Image" src="https://github.com/user-attachments/assets/3e7efbfd-645c-4020-bd98-d03da1e06b4d" />

</details>
